Stylelint issues for no-descending-specificity

Stylelint hasn't been running because of a config file issue, and these rules are introduced over a few PRs.
I've cleaned up a bunch in #1518, there there are some remaining issues for these rules that will be suppressed temporarily and tracked here

```
2020-09-07T19:12:20.2013871Z examples/carousel/css/carousel-prev-next.css
2020-09-07T19:12:20.2024859Z ##[error] 211:1  ✖  Expected selector ".carousel-moreaccessible .carousel-items .carousel-image a" to come before selector ".carousel .carousel-item .carousel-caption a:hover"                            no-descending-specificity
2020-09-07T19:12:20.2030693Z ##[error] 211:1  ✖  Expected selector ".carousel-moreaccessible .carousel-items .carousel-image a" to come before selector ".carousel .carousel-item .carousel-caption a:focus"                            no-descending-specificity
2020-09-07T19:12:20.2032118Z ##[error] 240:1  ✖  Expected selector ".carousel-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel .carousel-item .carousel-caption a:hover"                           no-descending-specificity
2020-09-07T19:12:20.2033339Z ##[error] 240:1  ✖  Expected selector ".carousel-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel .carousel-item .carousel-caption a:focus"                           no-descending-specificity
2020-09-07T19:12:20.2034550Z ##[error] 240:1  ✖  Expected selector ".carousel-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel-moreaccessible .carousel-items.focus .carousel-image a"             no-descending-specificity
2020-09-07T19:12:20.2035775Z ##[error] 250:1  ✖  Expected selector ".carousel-moreaccessible .carousel-item .carousel-caption span.contrast" to come before selector ".carousel .carousel-item .carousel-caption span.contrast:hover"   no-descending-specificity
2020-09-07T19:12:20.2037039Z ##[error] 280:1  ✖  Expected selector ".carousel-moreaccessible .controls" to come before selector ".carousel.carousel-moreaccessible .controls"                                                           no-descending-specificity
2020-09-07T19:12:20.2039008Z 
2020-09-07T19:12:20.2039505Z examples/carousel/css/carousel-tablist.css
2020-09-07T19:12:20.2040262Z ##[error] 272:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-items .carousel-image a" to come before selector ".carousel-tablist .carousel-item .carousel-caption a:hover"                            no-descending-specificity
2020-09-07T19:12:20.2041442Z ##[error] 272:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-items .carousel-image a" to come before selector ".carousel-tablist .carousel-item .carousel-caption a:focus"                            no-descending-specificity
2020-09-07T19:12:20.2042872Z ##[error] 301:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel-tablist .carousel-item .carousel-caption a:hover"                           no-descending-specificity
2020-09-07T19:12:20.2044074Z ##[error] 301:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel-tablist .carousel-item .carousel-caption a:focus"                           no-descending-specificity
2020-09-07T19:12:20.2045028Z ##[error] 301:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-item .carousel-caption a" to come before selector ".carousel-tablist-moreaccessible .carousel-items.focus .carousel-image a"             no-descending-specificity
2020-09-07T19:12:20.2045960Z ##[error] 311:1  ✖  Expected selector ".carousel-tablist-moreaccessible .carousel-item .carousel-caption span.contrast" to come before selector ".carousel-tablist .carousel-item .carousel-caption span.contrast:hover"   no-descending-specificity
2020-09-07T19:12:20.2047096Z ##[error] 354:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.border" to come before selector ".carousel-tablist [role="tab"]:focus circle.border"                                           no-descending-specificity
2020-09-07T19:12:20.2048104Z ##[error] 354:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.border" to come before selector ".carousel-tablist [role="tab"]:hover circle.border"                                           no-descending-specificity
2020-09-07T19:12:20.2049065Z ##[error] 358:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.tab" to come before selector ".carousel-tablist [role="tab"][aria-selected="true"] circle.tab"                                 no-descending-specificity
2020-09-07T19:12:20.2050683Z ##[error] 358:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.tab" to come before selector ".carousel-tablist [role="tab"]:hover circle.tab"                                                 no-descending-specificity
2020-09-07T19:12:20.2052148Z ##[error] 358:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.tab" to come before selector ".carousel-tablist [role="tab"][aria-selected="true"]:hover circle.tab"                           no-descending-specificity
2020-09-07T19:12:20.2053212Z ##[error] 362:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.tab-background" to come before selector ".carousel-tablist [role="tablist"].focus circle.tab-background"                       no-descending-specificity
2020-09-07T19:12:20.2054412Z ##[error] 362:1  ✖  Expected selector ".carousel-tablist-moreaccessible [role="tab"] circle.tab-background" to come before selector ".carousel-tablist [role="tab"]:hover circle.tab-background"                           no-descending-specificity
2020-09-07T19:12:20.2054768Z 
2020-09-07T19:12:20.2055368Z examples/combobox/css/combobox-datepicker.css
2020-09-07T19:12:20.2056293Z ##[error]  71:1  ✖  Expected selector ".combobox-datepicker .group.focus polygon" to come before selector ".combobox-datepicker .group button[aria-expanded="true"] polygon"   no-descending-specificity
2020-09-07T19:12:20.2057461Z ##[error] 112:1  ✖  Expected selector ".combobox-datepicker .dialog button" to come before selector ".combobox-datepicker .group.focus button"                                 no-descending-specificity
2020-09-07T19:12:20.2059349Z ##[error] 117:1  ✖  Expected selector ".combobox-datepicker .dialog button::-moz-focus-inner" to come before selector ".combobox-datepicker .group.focus button"               no-descending-specificity
2020-09-07T19:12:20.2060341Z ##[error] 165:1  ✖  Expected selector ".combobox-datepicker .dialog-ok-cancel-group button" to come before selector ".combobox-datepicker .group.focus button"                 no-descending-specificity
2020-09-07T19:12:20.2061811Z ##[error] 165:1  ✖  Expected selector ".combobox-datepicker .dialog-ok-cancel-group button" to come before selector ".combobox-datepicker .dialog button::-moz-focus-inner"    no-descending-specificity
2020-09-07T19:12:20.2062953Z ##[error] 237:1  ✖  Expected selector ".combobox-datepicker .dates td:focus" to come before selector ".combobox-datepicker .dates td:not(.disabled):hover"                     no-descending-specificity
2020-09-07T19:12:20.2063291Z 
2020-09-07T19:12:20.2063695Z examples/menubar/css/menubar-editor.css
2020-09-07T19:12:20.2064564Z ##[error] 104:1  ✖  Expected selector ".menubar-editor [role="menubar"] [role="separator"]" to come before selector ".menubar-editor [role="menubar"] [role="menu"] [role="separator"]"   no-descending-specificity
2020-09-07T19:12:20.2065520Z ##[error] 127:1  ✖  Expected no more than 1 empty line                                                                                                                                    max-empty-lines          
2020-09-07T19:12:20.2065878Z 
2020-09-07T19:12:20.2066291Z examples/menubar/css/menubar-navigation.css
2020-09-07T19:12:20.2067113Z ##[error] 36:1  ✖  Expected selector ".menubar-navigation [role="menuitem"]" to come before selector ".menubar-navigation [role="menu"] [role="menuitem"]"          no-descending-specificity
2020-09-07T19:12:20.2068120Z ##[error] 37:1  ✖  Expected selector ".menubar-navigation [role="separator"]" to come before selector ".menubar-navigation [role="menu"] [role="separator"]"        no-descending-specificity
2020-09-07T19:12:20.2069055Z ##[error] 68:1  ✖  Expected selector ".menubar-navigation > li > [role="menuitem"]" to come before selector ".menubar-navigation [role="menu"] [role="menuitem"]"   no-descending-specificity
2020-09-07T19:12:20.2069978Z ##[error] 88:1  ✖  Expected selector ".menubar-navigation [role="separator"]" to come before selector ".menubar-navigation [role="menu"] [role="separator"]"        no-descending-specificity
```

--- a/.stylelintrc
+++ b/.stylelintrc
@@ -9,7 +9,6 @@
         ]
       }
     ],
-    "no-descending-specificity": null,
     "prettier/prettier": [true, {"singleQuote": false}]
   }
 }

--- a/examples/carousel/css/carousel-prev-next.css
+++ b/examples/carousel/css/carousel-prev-next.css
@@ -63,6 +63,29 @@ img.reload {
   border: 0 solid transparent;
 }
 
+.carousel-moreaccessible .carousel-items .carousel-image a {
+  display: block;
+  margin: 0;
+  padding: 5px;
+  text-decoration: none;
+  border: none;
+}
+
+.carousel-moreaccessible .carousel-item .carousel-caption a {
+  display: inline-block;
+  margin: 0;
+  padding: 6px;
+  color: black;
+  background-color: transparent;
+  border: none;
+  border-radius: 5px;
+}
+
+.carousel-moreaccessible .carousel-item .carousel-caption span.contrast,
+.carousel-moreaccessible .carousel-item .carousel-caption span.contrast:hover {
+  background-color: transparent;
+}
+
 .carousel .carousel-item .carousel-caption a:hover,
 .carousel .carousel-item .carousel-caption span.contrast:hover {
   background-color: rgba(0, 0, 0, 1);
@@ -189,6 +212,14 @@ img.reload {
   border-radius: 5px;
 }
 
+/* Shared CSS for Pause and Tab Controls */
+
+.carousel-moreaccessible .controls {
+  position: relative;
+  top: 0;
+  left: 0;
+  padding: 0.25em 0.25em 0;
+}
 .carousel.carousel-moreaccessible .controls {
   position: static;
   height: 36px;
@@ -205,14 +236,6 @@ img.reload {
 .carousel-moreaccessible .carousel-items,
 .carousel-moreaccessible .carousel-items.focus {
   padding: 0;
-  border: none;
-}
-
-.carousel-moreaccessible .carousel-items .carousel-image a {
-  display: block;
-  margin: 0;
-  padding: 5px;
-  text-decoration: none;
   border: none;
 }
 
@@ -237,21 +260,6 @@ img.reload {
   color: black;
 }
 
-.carousel-moreaccessible .carousel-item .carousel-caption a {
-  display: inline-block;
-  margin: 0;
-  padding: 6px;
-  color: black;
-  background-color: transparent;
-  border: none;
-  border-radius: 5px;
-}
-
-.carousel-moreaccessible .carousel-item .carousel-caption span.contrast,
-.carousel-moreaccessible .carousel-item .carousel-caption span.contrast:hover {
-  background-color: transparent;
-}
-
 .carousel-moreaccessible .carousel-item .carousel-caption p {
   padding: 0;
   margin: 0;
@@ -273,13 +281,4 @@ img.reload {
   background-color: transparent;
   color: black;
   outline: none;
-}
-
-/* Shared CSS for Pause and Tab Controls */
-
-.carousel-moreaccessible .controls {
-  position: relative;
-  top: 0;
-  left: 0;
-  padding: 0.25em 0.25em 0;
 }

--- a/examples/carousel/css/carousel-tablist.css
+++ b/examples/carousel/css/carousel-tablist.css
@@ -63,6 +63,32 @@ img.reload {
   border: 0 solid transparent;
 }
 
+.carousel-tablist-moreaccessible .carousel-items .carousel-image a {
+  display: block;
+  margin: 0;
+  padding: 5px;
+  text-decoration: none;
+  border: none;
+}
+
+.carousel-tablist-moreaccessible .carousel-item .carousel-caption a {
+  display: inline-block;
+  margin: 0;
+  padding: 6px;
+  color: black;
+  background-color: transparent;
+  border: none;
+  border-radius: 5px;
+}
+
+.carousel-tablist-moreaccessible .carousel-item .carousel-caption span.contrast,
+.carousel-tablist-moreaccessible
+  .carousel-item
+  .carousel-caption
+  span.contrast:hover {
+  background-color: transparent;
+}
+
 .carousel-tablist .carousel-item .carousel-caption a:hover,
 .carousel-tablist .carousel-item .carousel-caption span.contrast:hover {
   background-color: rgba(0, 0, 0, 1);
@@ -221,8 +247,20 @@ img.reload {
   stroke-width: 2px;
 }
 
+.carousel-tablist-moreaccessible [role="tab"] circle.tab-background {
+  z-index: 16;
+}
+
+.carousel-tablist-moreaccessible [role="tab"] circle.tab {
+  z-index: 18;
+}
+
 .carousel-tablist [role="tab"][aria-selected="true"] circle.tab {
   fill: white;
+}
+
+.carousel-tablist-moreaccessible [role="tab"] circle.border {
+  z-index: 14;
 }
 
 .carousel-tablist [role="tab"]:focus circle.border {
@@ -269,14 +307,6 @@ img.reload {
   border: none;
 }
 
-.carousel-tablist-moreaccessible .carousel-items .carousel-image a {
-  display: block;
-  margin: 0;
-  padding: 5px;
-  text-decoration: none;
-  border: none;
-}
-
 .carousel-tablist-moreaccessible .carousel-items.focus .carousel-image a {
   padding: 2px;
   border: 3px solid #005a9c;
@@ -296,24 +326,6 @@ img.reload {
   margin: 0;
   height: 60px;
   color: black;
-}
-
-.carousel-tablist-moreaccessible .carousel-item .carousel-caption a {
-  display: inline-block;
-  margin: 0;
-  padding: 6px;
-  color: black;
-  background-color: transparent;
-  border: none;
-  border-radius: 5px;
-}
-
-.carousel-tablist-moreaccessible .carousel-item .carousel-caption span.contrast,
-.carousel-tablist-moreaccessible
-  .carousel-item
-  .carousel-caption
-  span.contrast:hover {
-  background-color: transparent;
 }
 
 .carousel-tablist-moreaccessible .carousel-item .carousel-caption p {
@@ -352,16 +364,4 @@ img.reload {
 
 .carousel-tablist-moreaccessible [role="tab"] {
   z-index: 20;
-}
-
-.carousel-tablist-moreaccessible [role="tab"] circle.border {
-  z-index: 14;
-}
-
-.carousel-tablist-moreaccessible [role="tab"] circle.tab {
-  z-index: 18;
-}
-
-.carousel-tablist-moreaccessible [role="tab"] circle.tab-background {
-  z-index: 16;
 }

--- a/examples/combobox/css/combobox-datepicker.css
+++ b/examples/combobox/css/combobox-datepicker.css
@@ -57,6 +57,26 @@
   outline-offset: 2px;
 }
 
+.combobox-datepicker .dialog button {
+  border-style: none;
+  background: transparent;
+}
+
+.combobox-datepicker .dialog-ok-cancel-group button {
+  padding: 6px;
+  margin-left: 1em;
+  width: 5em;
+  background-color: hsl(216, 80%, 92%);
+  font-size: 0.85em;
+  color: black;
+  outline: none;
+  border-radius: 5px;
+}
+
+.combobox-datepicker .dialog button::-moz-focus-inner {
+  border: 0;
+}
+
 .combobox-datepicker .group.focus input,
 .combobox-datepicker .group.focus button {
   background-color: #def;
@@ -67,8 +87,8 @@
   stroke: transparent;
 }
 
-.combobox-datepicker .group button[aria-expanded="true"] polygon,
-.combobox-datepicker .group.focus polygon {
+.combobox-datepicker .group.focus polygon,
+.combobox-datepicker .group button[aria-expanded="true"] polygon {
   fill: black;
   stroke: white;
 }
@@ -107,15 +127,6 @@
   color: white;
   text-transform: none;
   font-weight: bold;
-}
-
-.combobox-datepicker .dialog button {
-  border-style: none;
-  background: transparent;
-}
-
-.combobox-datepicker .dialog button::-moz-focus-inner {
-  border: 0;
 }
 
 .combobox-datepicker .dates {
@@ -160,17 +171,6 @@
   margin-top: 1em;
   margin-bottom: 1em;
   margin-right: 1em;
-}
-
-.combobox-datepicker .dialog-ok-cancel-group button {
-  padding: 6px;
-  margin-left: 1em;
-  width: 5em;
-  background-color: hsl(216, 80%, 92%);
-  font-size: 0.85em;
-  color: black;
-  outline: none;
-  border-radius: 5px;
 }
 
 .combobox-datepicker .dialog-button:focus {
@@ -229,15 +229,15 @@
   background-color: hsl(216, 80%, 92%);
 }
 
-.combobox-datepicker .dates td:not(.disabled):hover {
-  padding: 2px;
-  border: 1px solid rgb(100, 100, 100);
-}
-
 .combobox-datepicker .dates td:focus {
   padding: 1px;
   border: 2px solid rgb(100, 100, 100);
   outline: 0;
+}
+
+.combobox-datepicker .dates td:not(.disabled):hover {
+  padding: 2px;
+  border: 1px solid rgb(100, 100, 100);
 }
 
 .combobox-datepicker .dialog-message {

--- a/examples/menubar/css/menubar-editor.css
+++ b/examples/menubar/css/menubar-editor.css
@@ -78,6 +78,13 @@
   text-decoration: line-through;
 }
 
+.menubar-editor [role="menubar"] [role="separator"] {
+  padding-top: 3px;
+  background-image: url("../images/separator.svg");
+  background-position: center;
+  background-repeat: repeat-x;
+}
+
 .menubar-editor [role="menubar"] [role="menuitem"],
 .menubar-editor [role="menubar"] [role="menuitemcheckbox"],
 .menubar-editor [role="menubar"] [role="menuitemradio"],
@@ -102,13 +109,6 @@
 .menubar-editor [role="menubar"] [role="menu"] [role="separator"] {
   padding-left: 27px;
   width: 8em;
-}
-
-.menubar-editor [role="menubar"] [role="separator"] {
-  padding-top: 3px;
-  background-image: url("../images/separator.svg");
-  background-position: center;
-  background-repeat: repeat-x;
 }
 
 .menubar-editor [role="menubar"] [role="menu"] [aria-checked="true"] {

--- a/examples/menubar/css/menubar-navigation.css
+++ b/examples/menubar/css/menubar-navigation.css
@@ -64,6 +64,23 @@
   display: block;
 }
 
+.menubar-navigation > li > [role="menuitem"] {
+  display: inline-block;
+  margin: 2px;
+  padding: 4px;
+  padding-bottom: 8px;
+  background-color: #eee;
+  border: 0 solid #eee;
+  color: black;
+}
+
+.menubar-navigation [role="separator"] {
+  padding-top: 3px;
+  background-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' viewBox='0 0 12 12'%3E%3Cline x1='0' y1='6' x2='12' y2='6' style='stroke:black;stroke-width:1' /%3E%3C/svg%3E%0A");
+  background-position: center;
+  background-repeat: repeat-x;
+}
+
 .menubar-navigation [role="menu"] [role="menuitem"],
 .menubar-navigation [role="menu"] [role="separator"] {
   display: block;
@@ -71,16 +88,6 @@
   margin: 2px;
   padding: 4px;
   padding-left: 8px;
-  background-color: #eee;
-  border: 0 solid #eee;
-  color: black;
-}
-
-.menubar-navigation > li > [role="menuitem"] {
-  display: inline-block;
-  margin: 2px;
-  padding: 4px;
-  padding-bottom: 8px;
   background-color: #eee;
   border: 0 solid #eee;
   color: black;
@@ -124,20 +131,7 @@
   padding: 0;
 }
 
-.menubar-navigation [role="separator"] {
-  padding-top: 3px;
-  background-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' viewBox='0 0 12 12'%3E%3Cline x1='0' y1='6' x2='12' y2='6' style='stroke:black;stroke-width:1' /%3E%3C/svg%3E%0A");
-  background-position: center;
-  background-repeat: repeat-x;
-}
-
 /* aria-current styling */
-
-.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path,
-.menubar-navigation [role="menu"] [role="menuitem"][aria-current] {
-  padding-left: 4px;
-  border-left: 4px solid #034575;
-}
 
 .menubar-navigation > li > [role="menuitem"][aria-current],
 .menubar-navigation > li > [role="menuitem"].aria-current-path {
@@ -145,34 +139,17 @@
   border-bottom: 4px solid #034575;
 }
 
+.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path,
+.menubar-navigation [role="menu"] [role="menuitem"][aria-current] {
+  padding-left: 4px;
+  border-left: 4px solid #034575;
+}
+
 /* focus styling */
 
 .menubar-navigation.focus {
   padding: 0;
   border: #034575 solid 3px;
-}
-
-.menubar-navigation [role="menu"] [aria-expanded="true"],
-.menubar-navigation [role="menu"] [role="menuitem"]:focus,
-.menubar-navigation [role="menu"] [role="menuitem"]:hover {
-  outline: none;
-  background-color: #ccc;
-}
-
-.menubar-navigation [role="menu"] [role="menuitem"]:focus,
-.menubar-navigation [role="menu"] [role="menuitem"]:hover {
-  padding: 2px;
-  padding-left: 6px;
-  border: 2px solid #034575;
-}
-
-.menubar-navigation [role="menu"] [aria-expanded="true"].aria-current-path,
-.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path:focus,
-.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path:hover,
-.menubar-navigation [role="menu"] [role="menuitem"][aria-current]:focus,
-.menubar-navigation [role="menu"] [role="menuitem"][aria-current]:hover {
-  padding-left: 4px;
-  border-left: 4px solid #034575;
 }
 
 .menubar-navigation > li > [aria-expanded="true"],
@@ -189,6 +166,20 @@
   border: 2px solid #034575;
 }
 
+.menubar-navigation [role="menu"] [aria-expanded="true"],
+.menubar-navigation [role="menu"] [role="menuitem"]:focus,
+.menubar-navigation [role="menu"] [role="menuitem"]:hover {
+  outline: none;
+  background-color: #ccc;
+}
+
+.menubar-navigation [role="menu"] [role="menuitem"]:focus,
+.menubar-navigation [role="menu"] [role="menuitem"]:hover {
+  padding: 2px;
+  padding-left: 6px;
+  border: 2px solid #034575;
+}
+
 .menubar-navigation > li > [aria-expanded="true"].aria-current-path,
 .menubar-navigation > li > [role="menuitem"].aria-current-path:focus,
 .menubar-navigation > li > [role="menuitem"].aria-current-path:hover,
@@ -196,4 +187,13 @@
 .menubar-navigation > li > [role="menuitem"][aria-current]:hover {
   padding-bottom: 2px;
   border-bottom: 4px solid #034575;
+}
+
+.menubar-navigation [role="menu"] [aria-expanded="true"].aria-current-path,
+.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path:focus,
+.menubar-navigation [role="menu"] [role="menuitem"].aria-current-path:hover,
+.menubar-navigation [role="menu"] [role="menuitem"][aria-current]:focus,
+.menubar-navigation [role="menu"] [role="menuitem"][aria-current]:hover {
+  padding-left: 4px;
+  border-left: 4px solid #034575;
 }

